components/dashboard/src/data/organizations/org-settings-mutation.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { TeamSettings } from "@gitpod/gitpod-protocol";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { getGitpodService } from "../../service/service";
+import { getOrgSettingsQuery, TeamSettingsResult } from "./org-settings-query";
+
+type UpdateTeamSettingsArgs = Pick<TeamSettings, "workspaceSharingDisabled" | "teamId">;
+
+export const useUpdateTeamSettingsMutation = () => {
+    const queryClient = useQueryClient();
+
+    return useMutation<TeamSettings, Error, UpdateTeamSettingsArgs>({
+        mutationFn: async ({ teamId, workspaceSharingDisabled }) => {
+            return await getGitpodService().server.updateTeamSettings(teamId, { workspaceSharingDisabled });
```

Comment:
In an effort to keep more UI agnostic/unaware to orgs/teams - been trying to keep team/org id off as a param to mutations/queries and just have the mutation or query hook `useCurrentOrg()` internally so components don't have to pass it in. Example - https://github.com/gitpod-io/gitpod/blob/c02c4b1602eb07ca5cf5906b387db3b1f45e01b3/components/dashboard/src/data/billing-mode/org-billing-mode-query.ts#L15